src/osal/linux/osal.c

```diff
@@ -15,25 +15,26 @@
 
 #define _GNU_SOURCE
 
-#include <osal.h>
-#include <options.h>
-#include <log.h>
-
+#include <assert.h>
+#include <errno.h>
+#include <limits.h>
+#include <signal.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <time.h>
-#include <signal.h>
-#include <limits.h>
+#include <unistd.h>
 
+#include <arpa/inet.h>
+#include <net/if.h>
 #include <pthread.h>
+#include <sys/ioctl.h>
 #include <sys/syscall.h>
-#include <arpa/inet.h>
 
-#include <assert.h>
-#include <errno.h>
-#include <string.h>
-#include <unistd.h>
+#include <osal.h>
```

Comment:
It is good practice to include the header that declares the functions provided by a module first in that module. I.e. keep osal.h at the top. This helps ensure that the header is self-contained.

Reply:
Good point. I change to:

-     interface header
-     Headers from same project
-     Linux headers
-     Standard C headers

Sorted in each group